src/Routing/RouteBuilder.php

```diff
@@ -881,8 +887,9 @@ public function plugin(string $name, $options = [], $callback = null)
             $options = [];
         }
 
-        $params = ['plugin' => $name] + $this->_params;
         $path = $options['path'] ?? '/' . Inflector::dasherize($name);
+        unset($options['path']);
+        $params = ['plugin' => $name] + $options + $this->_params;
```

Comment:
This mixing of options and params is confusing and doesn't make for very readable code. The constructor has two separate arguments for params and options while for thos method now both seem to be clubbed together.

We should better document which keys options and params can contain and if possible avoid such clubbing. 

Also there's inconsistent in key naming. For e.g. for the constructor's options `namePrefix` doesn't have a underscore prefix while at other places it does.

Reply:
> We should better document which keys options and params can contain and if possible avoid such clubbing.

I considered copying over only the relevant keys but I got lazy. I can get unlazy if being explicit is important here.

> Also there's inconsistent in key naming. For e.g. for the constructor's options namePrefix doesn't have a underscore prefix while at other places it does.

`RouteBuilder::scope()` also uses `_namePrefix`. I'm pretty sure that option has an `_` because it was added in a minor release and we didn't want to shadow userland route keys.

Reply:
I pushed a commit to remove unneeded additional local var and array merging :slightly_smiling_face:.

Regarding the `$options` vs `$params` inconsistency/confusion; `plugin()`, `prefix()` are quite similar convenience wrappers for `scope()` which just preset certain values. Yet for `plugin()` the 2nd arg is called `$options` while for `prefix()` it's `$params` (which matches `scope()` when both args can have pretty much the same keys.

Reply:
That said I guess we can live we the current argument names.